static/js/WordSalad.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+
+export const WordSalad = ({ numLines, salad, renderItem }) => {
+/**
+ * This component renders a collection of text items, styled to fit within a specified
+ * number of lines. Each item can be rendered using a custom `renderItem` function, and
+ * spaces within the items are replaced with non-breaking spaces to ensure consistent
+ * line breaking behavior in CSS.
+ *
+ * @param {number} props.numLines - The number of lines the container should display.
+ *                                  This is used in the CSS to control layout.
+ * @param {Array} props.salad - An array of objects representing the text items to display.
+ *                              Each object must have a `text` property.
+ * @param {Function} props.renderItem - A function that receives an item from the `salad` array
+ *                                      and returns a React element to render it.
+ */
+
+  const nonWhitespaceInvisibleChar = '\u00A0'
+  // replace the normal space with the HTML space char, in order for css to not break line mid-item.
+  salad = salad.map(saladItem => ({
```

Comment:
Instead of doing this which requires the objects to have a `.text` field (which is problematic for bilingual text) you should be able to wrap each element in a span with the following CSS:

```css
.salad-item-no-wrap {
    white-space: nowrap;
}
```